api/opds2.py

```diff
@@ -58,3 +79,129 @@ def feed_links(self):
         return [
             {"href": self.url, "rel": "self", "type": self.OPDS2_TYPE},
         ]
+
+
+class OPDS2API(BaseCirculationAPI):
```

Comment:
Currently the WAYFLess post processor is only setup for OPDS and not OPDS2.0, so it does not actually come in the way right now, also since it is the post processor it should work as it does not after the `.fulfill()` method is called.
I agree with the repetition of code between CirculationAPI and OPDS2API, I was however under the impression that all cicrulation calls with an OPDS2 collection that has a token endpoint would require authentication with that endpoint, should I be making those changes directly in the CirculationAPI? Or should we not authenticate those calls at all?

Reply:
Thats good, I thought that the WAYFLess URL processor applied to OPDS 2.0 as well. 

> agree with the repetition of code between CirculationAPI and OPDS2API, I was however under the impression that all cicrulation calls with an OPDS2 collection that has a token endpoint would require authentication with that endpoint

There are three endpoints we call, I think: 
- The feed URL
- The authentication URL
- The fulfillment URL

The feed URL and the authentication URL don't require the token, only the fulfillment do. Are there any other endpoint calls I'm forgetting here?

Reply:
Yes, they do not require the token, I was thinking of it as an extra layer of proquest authorization ([example here](https://github.com/ThePalaceProject/circulation/pull/547/commits/35e8affce7591d78f3cf64fc3129097a62e629d0#diff-b8e9c8ebb72f45f352cf91d3f85f476e946a8bde5d29606216462092f47fcd76R125)). If we don't want to do that, its fine.